### PR TITLE
Added Migrating from an existing validator or contract to README

### PR DESCRIPTION
**Pull request by htafolla**

### CODE REVIEW

File: staking-pool/README.md [hunk 1]
```diff
@@ -298,3 +298,117 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+```bash
+./build.sh
+```
+#Get validator key:
+```bash
+cat ~/.near/betanet/validator_key.json |grep "public_key"
+```
+#Create a new account to deploy contract to
+- Set my_validator to the name you want publicly displayed
+- --mastAccount is your account you signed up to StakeWars2 with
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:360]:
```suggestion
- --masterAccount is your account you signed up to StakeWars2 with
```

File: staking-pool/README.md [hunk 2]
```diff
@@ -298,3 +298,117 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+near login
+```
+
+Unstake and Withdraw:
+```bash
+#If you staked to your validator unstake, there is no withdraw
+near stake nearkat.betanet <staking public key> 0
+
+#If you staked to a contract
+near call my_validator unstake '{"amount": "100000000000000000000000000"}' --accountId user1
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:322]: Add how to get the current amount from the contract:
`near view my_validator get_account_staked_balance '{"account_id": "user1"}'`

File: staking-pool/README.md [hunk 3]
```diff
@@ -298,3 +298,117 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+This provides instructions to migrate your staked validator or a validator contract to the new contract for Stakewars2
+
+Upgrade to the latest near-shell:
+```bash
+npm install -g near-shell
+```
+Login:
+```bash
+#If not logged into the browser recover your account with the seed phrase first
+export NEAR_ENV=betanet
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:312]: Explain what's this export line. E.g. If running on the `betanet`, switch the near-shell target network to `betanet`

File: staking-pool/README.md [hunk 4]
```diff
@@ -298,3 +298,117 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+#If you staked to a contract
+near call my_validator unstake '{"amount": "100000000000000000000000000"}' --accountId user1
+
+#Wait 3 epochs (9 hours) to withdraw and check if balance is available to withdraw
+near view my_validator is_account_unstaked_balance_available '{"account_id": "user1"}'
+
+#If is_account_unstaked_balance_available returns "true" withdraw
+near call my_validator withdraw '{"amount": "100000000000000000000000000"}' --accountId user1
+```
+#Download new contract with Git:
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:330]: Update headers size to `###` or `####`

File: staking-pool/README.md [hunk 5]
```diff
@@ -298,3 +298,117 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+
+#Login and authorize the delegator (complete the next 3 steps for each delegator account)
+```bash
+export NEAR_ENV=betanet
+near login
+```
+#Deposit to the valdiator
+```bash
+near call my_validator deposit '{}' --accountId user1 --amount 100
+near login
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:387]: remove `near login`

File: staking-pool/README.md [hunk 6]
```diff
@@ -298,3 +298,117 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+```
+#Stake the deposited amount
+```bash
+near call my_validator stake '{"amount": "100000000000000000000000000"}' --accountId user1
+near login
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:392]: same here

File: staking-pool/README.md [hunk 7]
```diff
@@ -298,3 +298,117 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+
+Unstake and Withdraw:
+```bash
+#If you staked to your validator unstake, there is no withdraw
+near stake nearkat.betanet <staking public key> 0
+
+#If you staked to a contract
+near call my_validator unstake '{"amount": "100000000000000000000000000"}' --accountId user1
+
+#Wait 3 epochs (9 hours) to withdraw and check if balance is available to withdraw
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:324]: 4 epochs now

File: staking-pool/README.md [hunk 8]
```diff
@@ -298,3 +298,124 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+##### Add wasm target to your toolchain:
+```bash
+rustup target add wasm32-unknown-unknown
+```
+#### Build:
+```bash
+./build.sh
+```
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:361]: BTW, this is not required, because the contract is already compiled and checked in. But if you want to recompile locally. These are the steps to follow.

Comment (htafolla) [reply staking-pool/README.md:361]: So we just need to have the checkout the contract and deploy it? I made several other slight revision, but can edit this section if you feel we should simplify.

Comment (evgenykuzyakov) [reply staking-pool/README.md:361]: I think we should keep, but add a comment to make it optional. In case folks don't want to deal with Rust and wasm32 targets.

File: staking-pool/README.md [hunk 9]
```diff
@@ -298,3 +298,140 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+#### Get validator key:
+```bash
+cat ~/.near/betanet/validator_key.json |grep "public_key"
+```
+#### Create a new account to deploy contract to
+- Set my_validator to the name you want publicly displayed
+- --masterAccount is your account you signed up to StakeWars2 with
+```bash
+near create_account my_validator --masterAccount=owner
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:374]: This step might fail because it's executed from the `staking-pool/initial-contracts/staking-pool` folder. It might not have `neardev` folder that you used to login.

I suggest to link it here using `ln -s ~/neardev neardev` (assuming you logged in from `~`)

File: staking-pool/README.md [hunk 10]
```diff
@@ -298,3 +298,140 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
...
+near validators next| grep "seat price"
+```
+#### Register a delegator account (repeat these steps for additional delegators)
+-- https://wallet.betanet.near.org
+-- backup your seed phrase
+-- transfer NEAR from your MasterAccount to the delegator account
+
+#### Login and authorize the delegator
+```bash
+export NEAR_ENV=betanet
```
Comment (evgenykuzyakov) [inline staking-pool/README.md:396]: Export is not needed for the 2nd time. It's already set to `betanet`

File: staking-pool/README.md (hunk 9 above)
Comment (evgenykuzyakov) [reply staking-pool/README.md:374]: This is old information. New `near-shell` stores everything under `~/.near-credentials/`, so disregard this comment.

File: staking-pool/README.md (hunk 10 above)
Comment (htafolla) [reply staking-pool/README.md:396]: Should I still work on the PR? Stefano had said he moved most of into StakeWars README. I still think it is viable and users are asking for more of a detailed approach.

Comment (evgenykuzyakov) [reply staking-pool/README.md:396]: Yep. Let's merge it here

File: staking-pool/README.md [hunk 11]
```diff
@@ -400,10 +400,6 @@ near login
 ```bash
 near call my_validator deposit '{}' --accountId user1 --amount 100
 ```
-#### Stake the deposited amount to the validator contract
```
Comment (evgenykuzyakov) [inline staking-pool/README.md]: Why did you remove this? It seems useful. When you just deposit, the amount is not staked immediately

Comment (htafolla) [reply staking-pool/README.md]: Bowen said the contract automatically stakes for you so you should not do it directly. Sorry for the confusion. I will fix that and get your feedback integrated. Only thing missing is creating the new node with the contract account. I will add it.
